spatialmedia/gui.py

```diff
@@ -39,73 +40,232 @@
         exit(0)
 
 
+class Console(object):
+    def __init__(self):
+        self.log = []
+
+    def append(self, text):
+        print text
+        self.log.append(text)
+
+
 class Application(Frame):
-    def open(self):
+    def action_open(self):
         """Triggers open file diaglog, reading a new file's metadata."""
-        self.in_file = tkFileDialog.askopenfilename(**self.open_options)
-        if self.in_file:
-            self.set_text("Opened file: %s\n" % self.in_file)
-            spherical.parse_metadata(self.in_file, self.append_text)
+        tmp_in_file = tkFileDialog.askopenfilename(**self.open_options)
+        if not tmp_in_file:
+            return
+        self.in_file = tmp_in_file
+
+        self.set_message("Opened file: %s\n" % ntpath.basename(self.in_file))
+
+        console = Console()
+        metadata = spherical.parse_metadata(self.in_file, console.append)
+
+        for line in console.log:
+            if "Error" in line:
+                self.set_error("Failed to load file %s"
+                               % ntpath.basename(self.in_file))
+                self.var_spherical.set(0)
+                self.disable_state()
+                self.button_open.configure(state="normal")
+                self.button_quit.configure(state="normal")
+                return
+
+        self.enable_state()
+        self.checkbox_spherical.configure(state="normal")
+
+        if not metadata:
+            self.set_message("No metadata found.")
 
-    def inject(self):
+        if metadata:
+            metadata = metadata.itervalues().next()
+            self.var_spherical.set(1)
+
+            self.set_message("Metadata found in %s\n"
+                             % ntpath.basename(self.in_file))
+            if metadata.get("Spherical", "") == "true":
```

Comment:
I'm guessing this isn't a boolean? Should we do the conversion to python primitives in the spherical module?

Also, this branch seems redundant from the default above (line 83). Please reconcile (remove line 83?)